static/panes/device-view.ts

```diff
@@ -0,0 +1,378 @@
+import * as monacoConfig from '../monaco-config';
+import {PaneRenaming} from '../widgets/pane-renaming';
+import TomSelect from 'tom-select';
+import GoldenLayout from 'golden-layout';
+import {Hub} from '../hub';
+import {MonacoPane} from './pane';
+import {DeviceAsmCode, DeviceAsmState} from './device-view.interfaces';
+import {MonacoPaneState} from './pane.interfaces';
+
+type DecorationEntry = {
+    linkedCode: any[];
+};
+
+type DeviceType = {
+    languageId: string;
+    asm: DeviceAsmCode[];
+};
+
+export class DeviceAsm extends MonacoPane<monaco.editor.IStandaloneCodeEditor, DeviceAsmState> {
+    private decorations: DecorationEntry;
+    private prevDecorations: string[];
+    private awaitingInitialResults: boolean;
+    private selectedDevice: string;
+    private devices: Record<string, DeviceType> | null;
+    private colours: string[];
+    private deviceCode: DeviceAsmCode[];
+    private lastColours: Record<number, number>;
+    private lastColourScheme: string;
+    private readonly selectize: TomSelect;
+    private linkedFadeTimeoutId: NodeJS.Timeout | null;
+
+    public constructor(hub: Hub, container: GoldenLayout.Container, state: DeviceAsmState & MonacoPaneState) {
+        super(hub, container, state);
+
+        this.prevDecorations = [];
+
+        this.awaitingInitialResults = false;
+        this.selection = state.selection;
+        this.selectedDevice = state.device || '';
+        this.devices = null;
+
+        this.colours = [];
+        this.deviceCode = [];
+        this.lastColours = [];
+        this.lastColourScheme = '';
+
+        const changeDeviceEl = this.domRoot[0].querySelector('.change-device') as HTMLInputElement;
+        this.selectize = new TomSelect(changeDeviceEl, {
+            sortField: 'name',
+            valueField: 'name',
+            labelField: 'name',
+            searchField: ['name'],
+            options: [],
+            items: [],
+            dropdownParent: 'body',
+            plugins: ['input_autogrow'],
+        });
+
+        this.paneRenaming = new PaneRenaming(this, state);
+
+        this.initButtons(state);
+        this.initEditorActions();
+
+        if (state.irOutput) {
+            this.showDeviceAsmResults(state.irOutput);
+        }
+    }
+
+    override getInitialHTML(): string {
+        return $('#device').html();
+    }
+
+    override createEditor(editorRoot: HTMLElement): monaco.editor.IStandaloneCodeEditor {
+        return monaco.editor.create(
+            editorRoot,
+            monacoConfig.extendConfig({
+                language: 'asm',
+                readOnly: true,
+                glyphMargin: true,
+                lineNumbersMinChars: 3,
+            })
+        );
+    }
+
+    override registerOpeningAnalyticsEvent(): void {
+        ga.proxy('send', {
+            hitType: 'event',
+            eventCategory: 'OpenViewPane',
+            eventAction: 'DeviceAsm',
+        });
+    }
+
+    initEditorActions(): void {
```

Comment:
This function should be `override registerEditorActions() ...` and will be called automatically from the base class constructor